### PR TITLE
Allow setting initial viewport before displaying image

### LINKED ISSUE #33: Add API's to support setting the initial viewport for an image without displaying it
See https://groups.google.com/forum/#!topic/cornerstone-platform/akHJ_mw2lzA

### PR DESCRIPTION
**Pull request by ClaireTagoe**
Fixes #33.

This change allows setViewport to set viewport on enabled elements with undefined images. Glad to make any changes you'd like, just let me know.

### CODE REVIEW

**Comment by ClaireTagoe** [comment]
Noticed the code climate issues. Please let me know what you think about fixing those.

**Comment by swederik** [comment]
Thanks for the PR! I'll try to get to this ASAP.

**Comment by ClaireTagoe** [comment]
friendly ping @swederik

File: src/setViewport.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 
 const MIN_WINDOW_WIDTH = 0.000001;
 const MIN_VIEWPORT_SCALE = 0.0001;
 
 /**
- * Sets the viewport for an element and corrects invalid values
+ * Sets/updates viewport of a given enabled element
  *
  * @param {HTMLElement} element - DOM element of the enabled element
  * @param {Viewport} viewport - Object containing the viewport properties
```
Comment (swederik) [inline src/setViewport.js]: If Viewport is now optional, could you update the jsdoc to reflect this? You just need square brackets around the parameter name.

e.g. * @param {Viewport} [viewport] - ...

File: src/setViewport.js [hunk 2]
```diff
@@ -14,32 +15,40 @@ const MIN_VIEWPORT_SCALE = 0.0001;
...
-  enabledElement.viewport.colormap = viewport.colormap;
-  enabledElement.viewport.labelmap = viewport.labelmap;
+  // If viewport is not already set, start with default and merge new
+  // viewport options later
+  if (enabledElement.viewport === undefined) {
+    enabledElement.viewport = getDefaultViewport(enabledElement.canvas);
+  }
+
+  // Merge viewport
+  if (viewport) {
```
Comment (swederik) [inline src/setViewport.js]: Could we do this with Object.assign instead? I'm not sure what it does with null / undefined values.

Comment (swederik) [reply src/setViewport.js]: Now that I said that, I see that displayImage does the same thing (https://github.com/cornerstonejs/cornerstone/blob/master/src/displayImage.js#L41) so you probably copied it from there. No need to change it I guess.

**Comment by swederik** [comment]
Could you add tiny test for the new functionality?
- setViewport should be able to set a viewport before an image is displayed

Then I guess it's good to merge.

**Comment by swederik** [comment]
Thanks!

**Pull request merged**